cmdstanpy/stanfit.py

```diff
@@ -1231,20 +1222,25 @@ class CmdStanGQ:
     Container for outputs from CmdStan generate_quantities run.
     """
 
-    def __init__(self, runset: RunSet, mcmc_sample: pd.DataFrame) -> None:
+    def __init__(
+        self,
+        runset: RunSet,
+        mcmc_sample: CmdStanMCMC,
+        logger: logging.Logger = None,
+    ) -> None:
         """Initialize object."""
         if not runset.method == Method.GENERATE_QUANTITIES:
             raise ValueError(
                 'Wrong runset method, expecting generate_quantities runset, '
                 'found method {}'.format(runset.method)
             )
         self.runset = runset
-        self._metadata = None
+        self._logger = logger or get_logger()
         self.mcmc_sample = mcmc_sample
         self._generated_quantities = None
-        self._column_names = scan_generated_quantities_csv(
-            self.runset.csv_files[0]
-        )['column_names']
+        self._generated_quantities_pd = None
```

Comment:
To stay consistent with what we just merged in #412, I recommend not making this None if you can avoid it. 

See: https://github.com/stan-dev/cmdstanpy/blob/fa339ab1709780589a0144c5d87afde550ad6b77/cmdstanpy/stanfit.py#L475-L478

You can't then use a test like `if self._generated_quantities_pd is None` unfortunately, but there are easy proxies of this (for numpy arrays, we never expect the shape to be equal to `(0,)` unless it is an unpopulated dummy array, for example)

Reply:
fixed